support/releaseinfo/registryclient/client.go

```diff
@@ -223,36 +225,98 @@ func ExtractImageFilesToDir(ctx context.Context, imageRef string, pullSecret []b
 }
 
 func getMetadata(ctx context.Context, imageRef string, pullSecret []byte) ([]distribution.Descriptor, distribution.BlobStore, error) {
+	repo, ref, err := GetRepoSetup(ctx, imageRef, pullSecret)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to repo setup: %w", err)
+	}
+	firstManifest, location, err := manifest.FirstManifest(ctx, ref, repo)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to obtain root manifest for %s: %w", imageRef, err)
+	}
+	_, layers, err := manifest.ManifestToImageConfig(ctx, firstManifest, repo.Blobs(ctx), location)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to obtain image layers for %s: %w", imageRef, err)
+	}
+	return layers, repo.Blobs(ctx), nil
+}
+
+func GetRepoSetup(ctx context.Context, imageRef string, pullSecret []byte) (distribution.Repository, reference.DockerImageReference, error) {
 	rt, err := rest.TransportFor(&rest.Config{})
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create secure transport: %w", err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to create secure transport: %w", err)
 	}
 	insecureRT, err := rest.TransportFor(&rest.Config{TLSClientConfig: rest.TLSClientConfig{Insecure: true}})
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create insecure transport: %w", err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to create insecure transport: %w", err)
 	}
 	credStore, err := dockercredentials.NewFromBytes(pullSecret)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to parse docker credentials: %w", err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to parse docker credentials: %w", err)
 	}
 	registryContext := registryclient.NewContext(rt, insecureRT).WithCredentials(credStore).
 		WithRequestModifiers(transport.NewHeaderRequestModifier(http.Header{http.CanonicalHeaderKey("User-Agent"): []string{rest.DefaultKubernetesUserAgent()}}))
 
 	ref, err := reference.Parse(imageRef)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to parse image reference %q: %w", imageRef, err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to parse image reference %q: %w", imageRef, err)
 	}
 	repo, err := registryContext.Repository(ctx, ref.DockerClientDefaults().RegistryURL(), ref.RepositoryName(), false)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create repository client for %s: %w", ref.DockerClientDefaults().RegistryURL(), err)
+		return nil, reference.DockerImageReference{}, fmt.Errorf("failed to create repository client for %s: %w", ref.DockerClientDefaults().RegistryURL(), err)
 	}
-	firstManifest, location, err := manifest.FirstManifest(ctx, ref, repo)
+	return repo, ref, nil
+}
+
+func IsMultiArchManifestList(ctx context.Context, imageRef string, pullSecret []byte) (bool, error) {
+	repo, ref, err := GetRepoSetup(ctx, imageRef, pullSecret)
+
```

Comment:
isn't this logic in findArchManifest as well? can this be a function

Reply:
Yeah it was. I moved it to a function.